File: inf-370-team-02-main/Sego and Bux API/Sego and Bux API/Models/ProductReview.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Sego_and__Bux.Models
{
    public class ProductReview
    {
        [Key]
        public int ReviewID { get; set; }

        [Required]
        public int UserID { get; set; }
        [ForeignKey("UserID")]
        public Customer Customer { get; set; }

        [Required]
        public int ProductID { get; set; }
        [ForeignKey("ProductID")]
        public Product Product { get; set; }

        [Required]
        public int OrderID { get; set; }

        [Required]
        [Range(1, 5)]
        public int Rating { get; set; }

        [MaxLength(200)]
        public string? ReviewTitle { get; set; }

        [Required]
        public string ReviewText { get; set; }

        [MaxLength(200)]
        public string? PhotoFileName { get; set; }

        [Required]
        public DateTime SubmittedDate { get; set; }

        [Required]
        [MaxLength(20)]
        public string Status { get; set; } = "Pending"; // Pending, Approved, Declined
    }
}

File: inf-370-team-02-main/sego-bux-shop/src/app/dto/product-review.ts
// src/app/dto/product-review.ts
export interface ProductReviewDto {
  reviewID: number;
  productID: number;
  userID: number;
  orderID: number;
  rating: number;
  reviewTitle?: string;
  reviewText: string;
  photoUrl?: string;
  submittedDate: string;
  status: string;
  userName: string;
  productName?: string; // if backend sends it we’ll use it
}

export interface CreateProductReviewDto {
  productID: number;
  orderID: number;
  rating: number;
  reviewTitle?: string;
  reviewText: string;
  photo?: File | null;
}
